cloudinit/distros/bsd_util.py

```diff
@@ -0,0 +1,36 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import shlex
+
+from cloudinit import util
+
+
+def get_rc_config_value(key, fn='/etc/rc.conf'):
+    contents = {}
+    for line in util.load_file(fn).splitlines():
+        if '=' in line:
+            k, v = line.split('=', 1)
+            contents[k] = v
+    return contents.get(key)
+
+
+def set_rc_config_value(key, value, fn='/etc/rc.conf'):
+    lines = []
+    done = False
+    value = shlex.quote(value)
+    for line in util.load_file(fn).splitlines():
```

Comment:
Does util.load_shell_content() work for you here? or  are there non-shell spaces that break parsing?